app/portainer/views/endpoints/edit/endpointController.js

```diff
@@ -34,6 +37,12 @@ function ($q, $scope, $state, $transition$, $filter, clipboard, EndpointService,
     $('#copyNotificationEdgeKey').show().fadeOut(2500);
   };
 
+  $scope.onCreateTag = async function onCreateTag(tagName) {
+    const tag = await TagService.createTag(tagName);
+    $scope.availableTags = $scope.availableTags.concat(tag);
+    $scope.endpoint.TagIds = $scope.endpoint.TagIds.concat(tag.Id);
+  }
```

Comment:
Wrap with non async function. The main goal for this wrapper is to trigger a digest cycle when the async function returns. See #2945 for more details.

Also, protect the await with a try/catch closure and show a notification if error occurs